xwiki-pro-macros-ui/src/main/resources/XWiki/Macros/Team.xml

```diff
@@ -656,50 +674,46 @@ $services.localization.render('rendering.macro.team.description')
-        #foreach($firstLetter in $name.toUpperCase().split("[\s-]"))$firstLetter.charAt(0)#end
+      &lt;span class="xwikiteam-avatar-initials-letters" style="color: $escapedLetterAvatarFontColor"&gt;
+        #foreach($firstLetter in $name.toUpperCase().split("[\s-]"))$escapetool.xml($firstLetter.charAt(0))#end
       &lt;/span&gt;
     &lt;/span&gt;
   &lt;/span&gt;
 #end
 #macro(avatar $username $size $disableLetterAvatars $letterAvatarBgColor $letterAvatarFontColor)
   #set($displayUser = $xwiki.getUserName($username, false))
-  #set($fontSize = $size / 2.5)
-  &lt;li class="xwikiteam-user" data-name-without-accents="${util.clearAccents($displayUser.toLowerCase())}" title="$displayUser" data-username="$username" style="font-size: ${fontSize}px"&gt;
-    #set($url = $xwiki.getURL($username))
-    #set($pictureListKey = "${size}-${username}")
-    #if(!$pictureList.containsKey($pictureListKey))
-      #set($pictureURL = '')
-      #set($profileDoc = $xwiki.getDocument($username))
-      #if(!$profileDoc.isNew())
-        #if($profileDoc.attachmentList.size() != 0)
-          #set($profileobj = $profileDoc.getObject('XWiki.XWikiUsers', 0))
-          #set($profileObjAvatar = $profileDoc.getValue('avatar'))
-          #if($profileObjAvatar &amp;&amp; $profileObjAvatar != '')
-            #foreach($attach in $profileDoc.attachmentList)
-              #if($attach.filename == $profileDoc.getValue('avatar'))
-                #set($pictureURL = $profileDoc.getAttachmentURL($profileObjAvatar, 'download', "height=$size"))
-                #break
-              #end
-            #end
-          #end
-        #end
-      #end
-      #if($pictureURL == "" &amp;&amp; !$disableLetterAvatars)
-        #set($picture = "#avatarInitials($displayUser $size $letterAvatarBgColor $letterAvatarFontColor)")
+  #set($escapedDisplayUser = $escapetool.xml($displayUser))
+  #set($fontSize = $size / 3)
+  &lt;li
+    class="xwikiteam-user"
+    data-name-without-accents="$util.clearAccents($escapedDisplayUser.toLowerCase())"
+    title="$escapedDisplayUser"
+    data-username="$escapetool.xml($username)"
+    style="font-size: ${fontSize}px"
+  &gt;
+    &lt;a href="$url"&gt;
+      #getUserAvatarURL($username $return $size)
+      #if(!$return.specified &amp;&amp; !$disableLetterAvatars)
+        #avatarInitials($displayUser $size $letterAvatarBgColor $letterAvatarFontColor)
       #else
-        #if($pictureURL == "")
-          #set($pictureURL = $xwiki.getAttachmentURL('XWiki.XWikiUserSheet', 'noavatar.png'))
-        #end
-        #set($picture = "&lt;img class='xwikiteam-avatar' src='$pictureURL' alt='$displayUser' title='$displayUser' width='$size' height='$size' /&gt;")
+        &lt;img
+          class='xwikiteam-avatar'
+          src='$escapetool.xml($return.url)'
+          alt='$escapedDisplayUser'
+          title='$escapedDisplayUser'
+          width='$size'
+          height='$size'
```

Comment:
Is the size (XML) safe? I see it is taken from configuration, so it could be anything. Why not escape its value?

Reply:
Fixed, I escaped the values.